docs/refman/loaderbase.py

```diff
@@ -94,8 +101,9 @@ def _validate_func(self, func: T.Union[Function, Method]) -> None:
             arg.type = self._resolve_type(arg.type.raw)
 
         # Handle returned_by
-        for obj in func.returns.resolved:
-            obj.data_type.returned_by += [func]
+        for obj in _Resolver._flatten_resolved(func.returns.resolved):
```

Comment:
I would have used
```suggestion
        for obj in self._flatten_resolved(func.returns.resolved):
```
but for what it's worth, I can get behind this PR. It seems `meson` is moving so fast that some sort of a team organization would be needed. I have found that the maintainer is responsive on the Matrix chat. Contacting them may help getting these changes in.